src/v/storage/segment_appender.cc

```diff
@@ -336,15 +336,23 @@ ss::future<> segment_appender::do_next_adaptive_fallocation() {
                  return _out.allocate(_fallocation_offset, step)
                    .then([this, step] { _fallocation_offset += step; });
              })
-      .handle_exception([this](std::exception_ptr e) {
-          vassert(
-            false,
+      .handle_exception_type([this](std::system_error e) {
+          vlog(
+            stlog.error,
             "We failed to fallocate file. This usually means we have ran out "
             "of disk space. Please check your data partition and ensure you "
             "have enough space. Error: {} - {}",
             e,
             *this);
+          ss::throw_with_backtrace<std::system_error>(e);
       });
+    // Would like to catch specific type above then the rest here.
```

Comment:
this comment is hard to read. are we or are we not catching errors. is this a `TODO(fabbri): ...`

Reply:
Yes.. comment for discussion to be removed before commit.

Reply:
I'm still getting "comfortable" with C++20.. Hoping to have a discussion about this.. then clean up this patch.